tuxemon/states/items/__init__.py

```diff
@@ -416,7 +416,7 @@ def buy_item(quantity: int) -> None:
             or not self.economy.lookup_item_price(item.slug)
             else self.economy.lookup_item_price(item.slug)
         )
-        money = self.buyer.game_variables.get("money", 0)
+        money = self.buyer.money.get("player")
         if int(price) == 0:
```

Comment:
If you are expecting the value to be there than use [] rather than get

Reply:
You're right, I didn't notice it.